.github/workflows/api-gateway.yml

```diff
@@ -0,0 +1,166 @@
+          [ "$VERSION" == "main" ] && VERSION=latest
+
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+
+          docker tag $API_GATEWAY_IMAGE_NAME $IMAGE_ID:$VERSION
+          docker push $IMAGE_ID:$VERSION
+
+  push_database:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      
+      - name: Build database image
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: ./db/build.sh 
+      
+      - name: Get database changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v6.3
+        with:
+          files: |
+            db/ 
+      - name: Log into registry
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}"  --password-stdin
+
+      - name: Push database image
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
+          IMAGE_ID=$REPO_NAME/$DATABASE_IMAGE_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+
+          docker tag $DATABASE_IMAGE_NAME $IMAGE_ID:$VERSION
+          docker push $IMAGE_ID:$VERSION
+  deploy:
+    needs:
+      - push_database
+      - push_api_gateway
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+    steps: 
+    - uses: actions/checkout@v2
+      with:
+          fetch-depth: 0
+
+    - name: Get all changed files
+      id: changed-files
+      uses: tj-actions/changed-files@v6.3
+      with:
+        files: |
+          db/
+          gateway/
+          docker-compose.yml
+
+    - name: Configure SSH
+      if: steps.changed-files.outputs.any_changed == 'true'
+      run: |
+        mkdir -p ~/.ssh/
+        echo "${{ secrets.SSH_PRIVATE_KEY }}" > ~/.ssh/staging.key
+        chmod 600 ~/.ssh/staging.key
+        cat >>~/.ssh/config <<END
+        Host staging
+          HostName ${{ secrets.WEB_CLIENT_HOST }}
+          User ec2-user
+          IdentityFile ~/.ssh/staging.key
+          StrictHostKeyChecking no
+        END
+       
+        # - name: Copy docker-compose files over SSH
+      # if: steps.changed-files.outputs.any_changed == 'true'
+      # run: |
+      # scp docker-compose.yml /home/ec2-user 
+      # scp docker-compose.prod.yml /home/ec2-user 
```

Comment:
just gonna manually copy these over to the EC2 instance since they rarely change and not sure if i have time to figure out whatever problem is going on with scp